<?xml version="1.0"?>

<launch>
      <!-- Start Gazebo with wg world running in (max) realtime -->
    <include file="$(find hector_gazebo_worlds)/launch/rolling_landscape_120m.launch"/>

    <!-- <include file="$(find multirobot_map_merge)/launch/map_merge.launch" /> -->

    <group ns="drone1">
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch" >
            <arg name="tf_prefix" value="drone1" />
            <arg name="name" value="drone1" />
            <arg name="x" value="-5.0" />
            <arg name="y" value="-20.0" />
            <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
            <arg name="controllers" value="
            controller/attitude
            controller/velocity
            controller/position
            "/>
        </include>
         <!-- Start SLAM system -->
        <include file="$(find hector_mapping)/launch/mapping_default.launch">
            <arg name="odom_frame" value="drone1/world"/>
            <arg name="base_frame" value="drone1/base_footprint"/>
            <arg name="pub_map_odom_transform" value="true"/>
            <arg name="scan_topic" value="drone1/scan"/>
            <arg name="map_frame" value="drone1/map"/>
        </include>

        <!-- Start GeoTIFF mapper -->
        <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch">
            <arg name="trajectory_publish_rate" value="4"/>
        </include>

        <include file="$(find two_drones)/launch/quadrotor_move_base.launch" >
            <arg name="base_frame_id"  value="/drone1/base_footprint"/>
            <arg name="odom_frame_id"  value="/drone1/world"/>
            <arg name="global_frame_id"  value="/drone1/map"/>
            <arg name="odom_topic" value="ground_truth/state" />
            <arg name="laser_topic" value="/drone1/scan" />
        </include>
    </group>

    <group ns="drone2">
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch" >
            <arg name="tf_prefix" value="drone2" />
            <arg name="name" value="drone2" />
            <arg name="x" value="-9.0" />
            <arg name="y" value="22.0" />
            <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
            <arg name="controllers" value="
            controller/attitude
            controller/velocity
            controller/position
            "/>
        </include>     
        <include file="$(find two_drones)/launch/quadrotor_gmapping.launch" >
            <arg name="scan_topic"  value="/drone2/scan" />
            <arg name="base_frame"  value="/drone2/base_footprint"/>
            <arg name="odom_frame"  value="/drone2/world"/>
        </include>
        <node pkg="tf" type="static_transform_publisher" name="map_to_drone2_broadcaster" args="0 0 0 0 0 0 map drone2/map 100"/>
        <include file="$(find two_drones)/launch/quadrotor_localization.launch" >
            <arg name="scan_topic"  value="/drone2/scan" />
            <arg name="base_frame_id"  value="/drone2/base_footprint"/>
            <arg name="odom_frame_id"  value="/drone2/world"/>
            <arg name="global_frame_id"  value="/drone2/map"/>
        </include>
        <include file="$(find two_drones)/launch/quadrotor_move_base.launch" >
            <arg name="base_frame_id"  value="/drone2/base_footprint"/>
            <arg name="odom_frame_id"  value="/drone2/world"/>
            <arg name="global_frame_id"  value="/drone2/map"/>
            <arg name="odom_topic" value="ground_truth/state" />
            <arg name="laser_topic" value="/drone2/scan" />
        </include>
    </group>
    

    <node pkg="two_drones" name="hover_node" type="hover_node" output="screen"/>

    <!-- Start rviz visualization with preset config -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hector_quadrotor_demo)/rviz_cfg/outdoor_two_drone_map_task1.rviz"/>

</launch>